### PR TITLE
Basic search functionality (Issue 3)

### PR DESCRIPTION
**Pull request by pete-a**
This PR aims to make a basic search engine for CoreWiki (Issue https://github.com/csharpfritz/CoreWiki/issues/3) that can be used as a foundation for more advanced search functionality later on.

My current search functionality is **VERY** basic. It simply does a case-insensitive search on the database to see if either the topic or content contains the query. Ordering is just by the Article view count DESC. I've implemented a `IArticlesSearchEngine` interface so it should be fairly easy to one day swap this basic search with something more advanced like ElasticSearch or Azure Search.

I also refactored the Article row from the All Articles page into a partial to keep the search page DRY. 

**Please note I've make a small UI tweak:** the Article row for 'All Articles' and the Article row for 'Latest Changes' were slightly different - the Latest Changes rows included a view count, whereas the All Articles rows did not. 

**TLDR:** 
- I've created basic search functionality that can be easily swapped out in the future via the `IArticlesSearchEngine` interface to something more advanced
- The All Articles page now shows the view count of each article, exactly like the Latest Changes page. If this is an issue let me know and I can change the Article Partial to show/hide the view count

### Items still on my todo list:
1. Pagination of the search results ✅ 
2. The links for each Article in article row don't work after the move to a partial for some reason ✅ 
    - This was due to slugs not being generated for Articles. E.g an unrelated bug that's now fixed on dev branch
3. Placing an inline search field in the nav bar. Bootstrap isn't playing nice and it just looked bad (see below). It looked even worse when the responsive layout kicked in. For now I've replaced it with simple link to the search page 
    - ❓ Maybe this UI improvement could be done in a future PR?

search-nav-bar
search-responsive

### CODE REVIEW

**Comment by DevOpsChallengeBr** [comment]
**Paging:** On the ALL Articles page has been implemented paging. What do you think about following the same paging technique?

**Comment by pete-a** [comment]
Thanks @DevOpsChallengeBr & @nmalocic.
I did a bit of research and couldn't find a built in `PagedCollection` (or similar) for ASP NET Core. [In the Microsoft docs](https://docs.microsoft.com/en-us/aspnet/core/data/ef-mvc/sort-filter-page?view=aspnetcore-2.1#add-paging-functionality-to-the-students-index-page) they seem to build their own pagination class too.

So I'll probably take a similar approach and also take advantage of the Pager TagHelper that was built for the All Articles page.

**Comment by csharpfritz** [comment]
This looks good... nice and simple search that gets the job done
